### PR TITLE
Convert testComponentInterface.cpp to use catch (#3555)

### PR DESCRIPTION
**Pull request by adamkewley**
Fixes issue #3555

The disabled test is disabled because the original code commented out the callsite in `main`.

- - -
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3563)

### CODE REVIEW

File: OpenSim/Common/Test/testComponentInterface.cpp [hunk 1]
```diff
@@ -1987,11 +2009,14 @@ void writeTimeSeriesTableForInputConnecteeSerialization() {
...
 }
 
-void testListInputConnecteeSerialization() {
+TEST_CASE("Component Interface List Input Connectee Serialization Behavior")
+{
     // We build a model, store the input connectee paths, then
     // recreate the same model from a serialization, and make sure the
     // connectee paths are the same.
 
+    writeTimeSeriesTableForInputConnecteeSerialization();
```
Comment (tkuchida) [inline OpenSim/Common/Test/testComponentInterface.cpp:2018]: Was this added intentionally? Was the test not working before?

File: OpenSim/Common/Test/testComponentInterface.cpp [hunk 2]
```diff
@@ -2085,10 +2110,12 @@ void testListInputConnecteeSerialization() {
...
 }
 
-void testSingleValueInputConnecteeSerialization() {
-
+TEST_CASE("Component Interface Single Value Input Connectee Serialization Behavior")
+{
     // Test normal behavior of single-value input (de)serialization.
     // -------------------------------------------------------------
+
+    writeTimeSeriesTableForInputConnecteeSerialization();
```
Comment (tkuchida) [inline OpenSim/Common/Test/testComponentInterface.cpp:2118]: Was this added intentionally?

**Review by tkuchida — COMMENTED** [review]
LGTM. Not sure whether 2 additions were intentional (commented below) or perhaps I'm misreading the diff. Also not sure what "The disabled test is disabled because the original code commented out the callsite in main." implies -- if the disabled test should be re-enabled at some point, is this tracked in an Issue? 🌳

File: OpenSim/Common/Test/testComponentInterface.cpp (hunk 2 above)
Comment (adamkewley) [reply OpenSim/Common/Test/testComponentInterface.cpp:2118]: (as above)

File: OpenSim/Common/Test/testComponentInterface.cpp (hunk 1 above)
Comment (adamkewley) [reply OpenSim/Common/Test/testComponentInterface.cpp:2018]: The original code placed the call in `main` before running those two tests.

The test content is dependent on the `sto` file generated by `writeTimeSeriesTableForInputConnecteeSerialization`, but the original version of the code didn't encode it into each case (instead, it relied on tests being ran in a certain order).

**Review by tkuchida — APPROVED** [review]

**Comment by adamkewley** [comment]
Thanks @tkuchida !

**Pull request merged**